pandas/io/parsers/readers.py

```diff
@@ -878,7 +939,7 @@ def _clean_options(self, options, engine):
         delim_whitespace = options["delim_whitespace"]
 
         if sep is None and not delim_whitespace:
-            if engine == "c":
+            if engine in ("c", "pyarrow"):
                 fallback_reason = (
                     "the 'c' engine does not support "
                     "sep=None with delim_whitespace=False"
```

Comment:
Might want to update the error message here to say `f"the '{engine}' does not support "` so that it refers to the specified engine.

Reply:
oops fixed now.